src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion},\
+ Weight: ${this.weight}`;
+  }
+
+  goForward(step = 1) {
+    this.coords.x += step;
+    this.coords.y += step;
+    this.coords.x -= step;
```

Comment:
```suggestion
    this.coords.y += step;
```
you add and subtract the same value. There is no sense to do it